### PR TITLE
ft(endpoint): create politica party

### PR DESCRIPTION
**Pull request by iamcristos**
#### What does this PR do?
endpoint that creates a new political party

#### Description of Task to be completed?
form inputed are passed, this input are first validated to ensure they are not emptied,
after validating a new political party is created

#### How should this be manually tested?
After cloning the reprository, run npm then test a post request on postman at '/api/v1/parties' ;
ensure to pass {name, hqaddress,logourl}. the id is generated automatically
#### What are the relevant pivotal tracker stories?

#163568731

#### Questions:

### CODE REVIEW

File: src/controllers/partyController.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import party from '../db/partyDb';
+
+class PartyController {
+  static createParty(req, res) {
+    const newParty = {
+      id: party.length + 1,
+      name: req.body.name,
+      hqaddress: req.body.hqaddress,
```
Comment (FredDominant) [inline src/controllers/partyController.js]: `hqaddress` should be in `camselCase` `hqAddress`

File: src/db/partyDb.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const party = [
+    {
```
Comment (FredDominant) [inline src/db/partyDb.js:2]: two parties shouldn't have the same `id`, or what do you think?

File: src/middlewares/partyMiddleware.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import validator from 'express-validator'
```
Comment (FredDominant) [inline src/middlewares/partyMiddleware.js]: I'm not sure you're supposed to use a `validator.` I think you're to write your validations yourself.

File: src/db/partyDb.js (hunk 2 above)
Comment (iamcristos) [reply src/db/partyDb.js:2]: Yes i think so to.. Thank you moving forward i will adjust them

File: src/controllers/partyController.js (hunk 1 above)
Comment (iamcristos) [reply src/controllers/partyController.js]: aiit tnx i will adjust it

File: src/middlewares/partyMiddleware.js (hunk 3 above)
Comment (iamcristos) [reply src/middlewares/partyMiddleware.js]: Aiit will write them

**Review by FredDominant — APPROVED** [review]

**Pull request merged**